### PR TITLE
Updated NAN to 2.14

### LINKED ISSUE #41: Nan API may have changed in Node 12
Recently, I started having Travis failures when using this lib on Node 12.
Starting from this line: https://travis-ci.org/comunica/comunica-actor-rdf-resolve-quad-pattern-hdt/jobs/527504675#L661

The same problem occurs in OSTRICH, which I still have to fix:
https://github.com/rdfostrich/ostrich-node/issues/3

Failing builds:
* https://github.com/comunica/comunica-actor-rdf-resolve-quad-pattern-hdt/issues/15
* https://github.com/comunica/comunica-actor-init-sparql-hdt/issues/66

### PR DESCRIPTION
**Pull request by mielvds**
Fixes #41 and makes HDT-Node compatible with Node 12. 
Not sure about the backwards compatibility now, to be tested.

### CODE REVIEW

**Comment by mielvds** [comment]
> Thanks! But can we also update `.travis.yml` to include `11`, `12`, `13`, `lts/*`, `node/*`? (cfr. https://github.com/rdfjs/N3.js/blob/7bbfb84e2d650edf896792c76ad7da1e5436fa3c/.travis.yml)

"node/*" or "node"?

File: lib/HdtDocument.cc [hunk 1]
```diff
@@ -66,7 +66,8 @@ const Nan::Persistent<Function>& HdtDocument::GetConstructor() {
     Nan::SetAccessor(constructorTemplate->PrototypeTemplate(),
                      Nan::New("closed").ToLocalChecked(), Closed);
     // Set constructor
-    constructor.Reset(constructorTemplate->GetFunction());
+//    constructor.Reset(constructorTemplate->GetFunction());
```
[the remark this answers is not in the document]
Comment (mielvds) [reply lib/HdtDocument.cc]: whops, yes

**Comment by RubenVerborgh** [comment]
Both.

**Comment by mielvds** [comment]
Ok, looks like they removed `Set` in Node 13.
Before it's:
```
../lib/HdtDocument.cc:209:21: warning: 'Set' is deprecated: Use maybe version [-Wdeprecated-declarations]
      triplesArray->Set(count++, tripleObject);
```

**Comment by RubenVerborgh** [comment]
Deprecation is fine for now; unless NAN already has an alternative.

**Comment by mielvds** [comment]
They do: https://github.com/nodejs/nan/blob/master/doc/maybe_types.md#nanset I'll see if it's much work

**Comment by mielvds** [comment]
latest commit enables building for 13 as well

**Comment by RubenVerborgh** [comment]
Okay the `node/*` was my bad; will fix. Looks good otherwise.

**Pull request merged**